src/jshint.js

```diff
@@ -274,23 +277,32 @@ var JSHINT = (function() {
     }
 
     if (state.option.worker) {
-      combine(predefined, vars.worker);
+      combine(predefined, globals.worker);
     }
 
     if (state.option.wsh) {
-      combine(predefined, vars.wsh);
-    }
-
-    if (state.option.globalstrict && state.option.strict !== false) {
-      state.option.strict = true;
+      combine(predefined, globals.wsh);
     }
 
     if (state.option.yui) {
-      combine(predefined, vars.yui);
+      combine(predefined, globals.yui);
     }
 
     if (state.option.mocha) {
-      combine(predefined, vars.mocha);
+      combine(predefined, globals.mocha);
+    }
+
+    if (state.option.nonstandard) {
+      combine(predefined, globals.nonstandard);
```

Comment:
I'm a little concerned about the dependency and the fact jshint would no longer be in control.
Case in point: nonstandard has been merged into builtin in the latest version - https://github.com/sindresorhus/globals/pull/36/files

So, yes, it would be nice, but once we switch we have to either slowly put things back if they break or else release a new major of jshint with breaking changes. So you can understand its a bit scary.

This is going to need the opinion of (another more) core team member I think.